lib/Doctrine/DBAL/Schema/Visitor/DropSchemaSqlCollector.php

```diff
@@ -82,19 +83,19 @@ public function getQueries()
     {
         $sql = [];
 
-        /** @var ForeignKeyConstraint $fkConstraint */
         foreach ($this->constraints as $fkConstraint) {
+            assert($fkConstraint instanceof ForeignKeyConstraint);
```

Comment:
Not necessarily for now, but would it be possible to document the property as `SplObjectStorage<ForeignKeyConstraint>` and avoid runtime assertions? I may have tried earlier but not sure.

Reply:
Documenting it as `SplObjectStorage<ForeignKeyConstraint, Table>` makes Psalm happy, but Phpstan gives the following error: `PHPDoc tag @var for property                                                                   
         Doctrine\DBAL\Schema\Visitor\DropSchemaSqlCollector::$constraints contains unresolvable type.`

Reply:
Alternatively, it might be something like `SplObjectStorage&Iterable<ForeignKeyConstraint>`.